lib/txmgr/txmgr_internal_test.go

```diff
@@ -218,6 +218,9 @@ func (b *mockBackend) HeaderByNumber(ctx context.Context, number *big.Int) (*typ
 		num.Set(number)
 	}
 
+	b.g.mu.RLock()
```

Comment:
prefer to access shared resources through getters, instead of havings user wrangle locks themselves.

```
func (fooer) SetFoo(foo)
f.mu.lock
defer f.mu.unlock
f.foo = foo
}

func (fooer) GetFoo()
f.mu.lock
defer f.mu.unlock
return f.foo
}

```